### PR TITLE
Update `CallbackFilterIterator` stub.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: stubs/iterable.stub [hunk 1]
```diff
@@ -206,21 +206,35 @@ class FilterIterator extends IteratorIterator
...
+    /**
+     * @param Iterator<TKey, TValue> $iterator
+     * @param callable(TValue, TKey, Iterator<TKey, TValue>): bool $callback
+     */
+    public function __construct(Iterator $iterator, callable $callback) {}
 
+    /**
+     * @return TValue|null
+     */
+    public function current() {}
```
Comment (drupol) [inline stubs/iterable.stub]: @ondrejmirtes Should I add this method? WDYT? (same for the `key` method).

**Comment by ondrejmirtes** [comment]
Why are you making these changes?

**Comment by drupol** [comment]
After some more thinking, I think those methods are not relevant because we are inheriting from `FilterIterator`. Going to make a commit to remove them: https://phpstan.org/r/159f7749-5b6b-409f-939e-958879348c95

**Comment by ondrejmirtes** [comment]
Why are you making these changes? Why the iterators shouldn't have covariant template types?

**Comment by drupol** [comment]
@ondrejmirtes Without the `-covariant` suffix, this is failing.

What do you advise here?

**Comment by arnaud-lb** [comment]
> /cc @arnaud-lb I have a question for you - am I right that @template-covariant parameter in a constructor is not a problem and should not be reported? Here's a reproduction: https://phpstan.org/r/f5a5d99b-a62b-4c81-8325-09984d8cef43

Agreed, this should not be reported for the constructor

**Comment by ondrejmirtes** [comment]
Hi, I'm cleaning up old and stale PRs. Please send a new PR if you're still interested, thanks.

**Pull request closed without merging**